crates/router/src/services/api.rs

```diff
@@ -164,7 +164,8 @@ where
         payments::CallConnectorAction::Trigger => {
             match connector_integration.build_request(req, &state.conf.connectors)? {
                 Some(request) => {
-                    let response = call_connector_api(state, request).await;
+                    let response = call_connector_api(state, request.clone()).await;
```

Comment:
We need actual request what we are sending to the connector...It will be easy to find the issues later. 
As you said payload is in secret, We need to change that. Only certain fields can be secret not the entire payload.

Reply:
The logs can always be searched by their respective keys, having them in the same place is not necessary. This can help us avoid the unnecessary clones. 

If you plan to take up the task of removing secret from payload in a separate PR, please mention the issue here.

Reply:
Removing secret from payload will be fixed by this #308.